src/software/ai/navigator/obstacle/obstacle_factory.cpp

```diff
@@ -1,101 +1,172 @@
 #include "software/ai/navigator/obstacle/obstacle_factory.h"
 
 ObstacleFactory::ObstacleFactory(std::shared_ptr<const ObstacleFactoryConfig> config)
-    : config(config)
+    : config(config),
+      shape_expansion_amount(config->RobotObstacleInflationFactor()->value() *
+                             ROBOT_MAX_RADIUS_METERS)
 {
 }
 
-std::vector<Obstacle> ObstacleFactory::getObstaclesFromMotionConstraints(
-    const std::set<MotionConstraint> &motion_constraints, const World &world)
+std::vector<ObstaclePtr> ObstacleFactory::createObstaclesFromMotionConstraint(
+    const MotionConstraint &motion_constraint, const World &world)
 {
-    std::vector<Obstacle> obstacles;
-    Rectangle rectangle({0, 0}, {1, 1});
-    for (auto motion_constraint : motion_constraints)
+    std::vector<ObstaclePtr> obstacles;
+    std::optional<Circle> circle_opt       = std::nullopt;
+    std::optional<Rectangle> rectangle_opt = std::nullopt;
+
+    switch (motion_constraint)
     {
-        switch (motion_constraint)
+        case MotionConstraint::ENEMY_ROBOTS_COLLISION:
         {
-            case MotionConstraint::ENEMY_ROBOTS_COLLISION:
-            {
-                std::vector<Obstacle> enemy_robot_obstacles =
-                    getVelocityObstaclesFromTeam(world.enemyTeam());
-                obstacles.insert(obstacles.end(), enemy_robot_obstacles.begin(),
-                                 enemy_robot_obstacles.end());
-            }
+            std::vector<ObstaclePtr> enemy_robot_obstacles =
+                createVelocityObstaclesFromTeam(world.enemyTeam());
+            obstacles.insert(obstacles.end(), enemy_robot_obstacles.begin(),
+                             enemy_robot_obstacles.end());
+        }
+        break;
+        case MotionConstraint::CENTER_CIRCLE:
+            obstacles.push_back(createObstacle(
+                Circle(world.field().centerPoint(),
+                       world.field().centerCircleRadius() + shape_expansion_amount)));
+            break;
+        case MotionConstraint::HALF_METER_AROUND_BALL:
+            // 0.5 represents half a metre radius
+            obstacles.push_back(createObstacle(
+                Circle(world.ball().position(), 0.5 + shape_expansion_amount)));
```

Comment:
Hey Joe! 
Ideally, we want to avoid having any circle_opt/rectangle_opt, but we can't get rid of it until we fix how we expand obstacles, which is out of scope for this PR